Add reason for enabling XCOM pickling (#747)
astronomer/astro-sdk: README.md

    airflow db init
    ```

>     > **Note:** `AIRFLOW__CORE__ENABLE_XCOM_PICKLING` needs to be enabled for `astro-sdk-python`. Currently, Custom XCom Backend cannot serialize all the `types` correctly. `AIRFLOW__CORE__ENABLE_XCOM_PICKLING` is needed for Dataframe and any other classes or types that aren’t JSON or your XCom backend serializable in airflow. The data format used by pickle is Python-specific. This has the advantage that there are no restrictions imposed by external standards such as JSON or XDR (which can’t represent pointer sharing); however it means that non-Python programs may not be able to reconstruct pickled Python objects. Read more: [enable_xcom_pickling](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#enable-xcom-pickling) and [pickle](https://docs.python.org/3/library/pickle.html#comparison-with-json):

Reviewer: `classes or types that aren’t JSON or your XCom backend serializable in airflow.` -- Feels incomplete?

Author: Added the change

>     > **Note:** `AIRFLOW__CORE__ENABLE_XCOM_PICKLING` needs to be enabled for `astro-sdk-python`. Currently, Custom XCom Backend cannot serialize all the `types` correctly. `AIRFLOW__CORE__ENABLE_XCOM_PICKLING` is needed for Dataframe and any other classes or types that aren’t JSON or your XCom backend serializable in airflow. The data format used by pickle is Python-specific. This has the advantage that there are no restrictions imposed by external standards such as JSON or XDR (which can’t represent pointer sharing); however it means that non-Python programs may not be able to reconstruct pickled Python objects. Read more: [enable_xcom_pickling](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#enable-xcom-pickling) and [pickle](https://docs.python.org/3/library/pickle.html#comparison-with-json):

Reviewer 2: ```suggestion
    > **Note:** `AIRFLOW__CORE__ENABLE_XCOM_PICKLING` needs to be enabled for `astro-sdk-python`. 
    
 Currently, custom XCom backends are limited to data types that are json serializable. Since Dataframes are not json serializable, we need to enable XCom pickling to store dataframes. 
  
The data format used by pickle is Python-specific. This has the advantage that there are no restrictions imposed by external standards such as JSON or XDR (which can’t represent pointer sharing); however it means that non-Python programs may not be able to reconstruct pickled Python objects. 

Read more: [enable_xcom_pickling](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#enable-xcom-pickling) and [pickle](https://docs.python.org/3/library/pickle.html#comparison-with-json):
```

Reviewer 2: Let's split this into multiple paragraphs as this is a REALLY dense chunk of information.

    airflow db init
    ```

    > **Note:** `AIRFLOW__CORE__ENABLE_XCOM_PICKLING` needs to be enabled for `astro-sdk-python`.

 Currently, custom XCom backends are limited to data types that are json serializable. Since Dataframes are not json serializable, we need to enable XCom pickling to store dataframes.

> The data format used by pickle is Python-specific. This has the advantage that there are no restrictions imposed by external standards such as JSON or XDR (which can’t represent pointer sharing); however it means that non-Python programs may not be able to reconstruct pickled Python objects.

Reviewer: Do we need this?

Author: It would add in context to user for why we need this.

Reviewer: >however it means that non-Python programs may not be able to reconstruct pickled Python objects.

I mean this line, we don't need that one ^^. But I will let you decide

Reviewer 3: Maybe if someone wants to opt-out and use airflow rest api?

Author: I did not understand @feluelle. Could you elaborate on this, please?

Reviewer 3: If you read data from xcom via the airflow rest api you would not be able to reconstruct it or? Maybe only with a python rest api client - but not the go client for example or? (that is what the sentence is saying, right? So maybe it is useful to leave it there.)

    airflow db init
    ```

    > **Note:** `AIRFLOW__CORE__ENABLE_XCOM_PICKLING` needs to be enabled for `astro-sdk-python`.

    Currently, custom XCom backends are limited to data types that are json serializable. Since Dataframes are not json serializable, we need to enable XCom pickling to store dataframes.

    The data format used by pickle is Python-specific. This has the advantage that there are no restrictions imposed by external standards such as JSON or XDR (which can’t represent pointer sharing); however it means that non-Python programs may not be able to reconstruct pickled Python objects.

    Read more: [enable_xcom_pickling](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#enable-xcom-pickling) and [pickle](https://docs.python.org/3/library/pickle.html#comparison-with-json):
            ```shell
            export AIRFLOW__CORE__ENABLE_XCOM_PICKLING=True

>             ```

Reviewer: Do we need this since we already set this on L47

Author: Removed it
